src/Lean/Data/Json/Stream.lean

```diff
@@ -9,19 +9,18 @@ import Lean.Data.Json.Parser
 import Lean.Data.Json.Printer
 import Lean.Data.Json.FromToJson
 
-namespace IO.FS.Handle
+namespace IO.FS.Stream
 
 open Lean
 open IO
 
-def readJson (h : FS.Handle) (nBytes : Nat) : IO Json := do
-bytes ← h.read nBytes;
-some s ← pure bytes.utf8ToString
-  | throw (userError ("got non-utf8 string '" ++ toString bytes ++ "'"));
+def readJson (h : FS.Stream) (nBytes : Nat) : IO Json := do
+bytes ← h.read (USize.ofNat nBytes);
+let s := String.fromUTF8Unchecked bytes;
```

Comment:
Is this safe on arbitrary data from client?

Reply:
As far as I know, yes: https://github.com/leanprover/lean4/pull/167#discussion_r479012793
What the current implementation actually does it to return the string as is, with invalid bytes counted as one codepoint each. But guarding against a malicious attacker is a bit moot anyway when we go on to execute arbitrary user code in the server.